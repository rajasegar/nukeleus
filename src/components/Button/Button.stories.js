import React from 'react';
import { select, boolean } from '@storybook/addon-knobs';

import { Button } from './index';

export default {
  component: Button,
  title: 'Button',
  argTypes: {
    backgroundColor: { control: 'color' },
  },
};

export const Basic = () => {
  const propsWithKnobs = {
    variant: select('variant', ['secondary', 'primary', 'danger','transparent']),
    disabled: boolean('disabled', false),
    size: select('scale', ['normal','mini'])
  };

  return <Button {...propsWithKnobs}>New Ticket</Button>;
};
const Template = (args) => <Button {...args} >Click me</Button>;

export const Primary = Template.bind({});
Primary.args = {
  variant: 'primary',
  label: 'Button',
};

export const Secondary = Template.bind({});
Secondary.args = {
  label: 'Button',
  variant: 'secondary',
};


export const PrimaryMini = Template.bind({});
PrimaryMini.args = {
  variant: 'primary',
  size: 'mini'
};
