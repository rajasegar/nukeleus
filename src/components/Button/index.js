// TODO: Later need to change it as styled-system - variant
import styled from 'styled-components';
import PropTypes from 'prop-types';
import {
  layout,
  space,
  position,
  background,
  border,
  typography,
  color,
  flexbox,
  shadow,
} from 'styled-system';

const disabledState = {
  primary: {
    background: 'linear-gradient(180deg, #92A2B1 4.39%, #7B8E9F 94.46%)',
    border: '1px solid #7B8E9F',
  },
  secondary: {
    background: '#EBEFF3',
    border: '1px solid #CFD7DF',
  },
  danger: {
    background: 'linear-gradient(180deg, #F89FA1 2.56%, #F2797B 95.75%)',
    border: '1px solid #F2797B',
  },
};

export const Button = styled('button')(
  ({
    variant = 'secondary',
    width = 80,
    mr = '',
    iconOnly = false,
    theme: { colors, gradients, shadows },
    customStyle,
    hoverBg,
    size,
  }) => ({
    cursor: 'pointer',
    minWidth: width,
    outline: 'none',
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    background: gradients[variant],
    border: `${variant === 'transparent' ? 'none' : '1px solid'}`,
    borderColor: `${
      variant === 'secondary'
        ? colors.input.border
        : variant === 'transparent'
        ? 'none'
        : colors.button[variant]
    }`,
    borderRadius: 4,
    boxShadow: `${variant === 'transparent' ? 'none' : shadows.input}`,
    color: variant === 'secondary' ? colors.button.primary : colors.white,
    fontSize: 14,
    fontWeight: 500,
    height: `${size === 'mini' ? '24px' : '32px'}`,
    marginRight: mr,
    padding: `${iconOnly ? '4px' : '6px 12px'}`,
    '&:disabled': {
      opacity: 0.4,
      pointerEvents: 'none',
      cursor: 'not-allowed',
      ...disabledState[variant],
    },
    '&:focus': {
      border: `1px solid ${colors.input.focus}`,
      boxShadow: `0 0 0 1px ${colors.input.focus}`,
    },
    '&:hover': {
      background: hoverBg ? hoverBg : colors.button[variant],
    },
    svg: {
      marginRight: `${iconOnly ? '0' : '7px'}`,
    },
    ...customStyle,
  }),
  space,
  layout,
  position,
  background,
  border,
  typography,
  color,
  flexbox,
  shadow
);

Button.defaultProps = {
  variant: 'secondary',
  size: 'normal',
};

Button.propTypes = {
  variant: PropTypes.oneOf(['secondary', 'primary', 'danger', 'transparent']),
  size: PropTypes.oneOf(['normal','mini'])
};

export const LoadingButton = styled(Button)`
  min-width: ${(props) => props.width || '80px'};
  .loader {
    width: 16px;
    height: 16px;
    animation: spin 1s linear infinite;

    svg {
      path {
        fill: ${(props) => props.theme.colors.primary};
      }
    }
  }

  @keyframes spin {
    from {
      transform: rotate(0deg);
    }
    to {
      transform: rotate(360deg);
    }
  }
`;

export default Button;
