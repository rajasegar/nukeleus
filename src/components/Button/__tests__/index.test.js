import React from 'react';
import { renderWithTheme } from '../../../test-utils';

import Button from '../index';

describe('Button component test suite', () => {
  it('should render Button', () => {
    const { getByText } = renderWithTheme(
      <Button variant="primary">Button</Button>
    );
    expect(getByText('Button')).toMatchSnapshot();
  });
});
