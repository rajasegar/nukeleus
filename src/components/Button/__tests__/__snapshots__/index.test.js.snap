// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component test suite should render Button 1`] = `
.c0 {
  outline: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: linear-gradient(180deg,#264966 0%,#12344D 100%);
  border: 1px solid #12344D;
  border-radius: 4px;
  box-shadow: inset 0px 1px 2px rgba(24,50,71,0.05);
  color: #FFFFFF;
  font-size: 14px;
  font-weight: 500;
  height: 32px;
  padding: 8px 12px;
  min-width: 80px;
}

.c0:disabled {
  opacity: 0.4;
}

.c0:focus {
  border: 2px solid #2C5CC5;
}

.c0:hover {
  background: #12344D;
  cursor: pointer;
}

<button
  class="c0"
>
  Button
</button>
`;
